### ISSUE TITLE
Generate project details for existing project

### ISSUE THREAD

**Issue by seadowg** [getodk/collect#4579]
## Acceptance

- [x] Given I've upgraded the app from an older version
When I go to the projects settings screen
Then the current project name has been generated based on the project's server
And the current project icon is the first letter of the project name

## Note

The generated project name should work the same way as the "Add project" form.

**Comment by seadowg** [comment]
@lognaturel we could possibly give everyone an emoji icon (either a fixed one or a random one from a small set) to make it clear that you can use emojis as the icon. Definitely a bit of extra delight but potentially too playful?

**Comment by lognaturel** [comment]
Project name should be the server's full domain name including any subdomains but no paths:
- `https://test.getodk.org/v1/key/jfkdlsajfaioehafnelsajfkdsljfs;jfsa/projects/17` => `test.getodk.org`
- `https://demo.getodk.org` => `demo.getodk.org`
- `https://ona.io/somebody` => `ona.io`

> give everyone an emoji icon

I don't think users will typically change their defaults and we want the default to provide some kind of value. I think it's important to use a letter for that reason:
- `test.getodk.org` => `T`
- `demo.getodk.org` => `D`
- `ona.io` => O

Is default color captured somewhere else?

**Comment by seadowg** [comment]
> Project name should be the server's full domain name including any subdomains but no paths

That's actually a little different from how we described it in https://github.com/getodk/collect/issues/4428. Easy to change though. Maybe better to discuss there as it's where we're first adding the server URL to name mapping in.

> I don't think users will typically change their defaults and we want the default to provide some kind of value.

Yeah convinced! The emoji is nice looking but would be fairly useless for them.

> Is default color captured somewhere else?

I posted another question about that in https://github.com/getodk/collect/issues/4493.
